### PR TITLE
feat: add parseSilentBlock

### PR DESCRIPTION
**Pull request by chaitika**

### CODE REVIEW

File: packages/core/src/encoding.ts [hunk 1]
```diff
@@ -63,3 +69,41 @@ export const createLabeledSilentPaymentAddress = (
 const hrpFromNetwork = (network: Network): string => {
     return network.bech32 === 'bc' ? 'sp' : 'tsp';
 };
+
+export const parseSilentBlock = (data: Buffer): SilentBlock => {
+    const type = data.readUInt8(0);    // read 8 bits from 0th index // unsigned integer - UInt
```
Comment (theanmolsharma) [inline packages/core/src/encoding.ts:74]: I think the method `readUInt8` is self-explanatory and doesn't warrant a comment explaining what it does.

```suggestion
    const type = data.readUInt8(0);
```

File: packages/core/test/fixtures/encoding.ts [hunk 2]
```diff
@@ -72,3 +72,44 @@ export const labelled = [
...
+                    },
+                    {
+                        pubKey: 'e7a3ab33572f47b074878c749a5f9a5d7e026f10b6ffcab1388e9356c07828fa',
+                        value: 4400,
+                        vout: 1,
+                    },
+                ],
+                isSpent: false,
+            },
+        ],
```
Comment (theanmolsharma) [inline packages/core/test/fixtures/encoding.ts:111]:
```suggestion
        parsedBlock: {
            type: 0,
            transactions: [
                {
                    txid: 'd60cdcc1bdb78d44bf83e2ae0efa94bbb1187ba2397e4a40960d853b78e15b4f',
                    scanTweak:
                        '0381a4ef11a26108cf10d8c33305b77aec009beed82a77bb47b716a04ae9ae5bf1',
                    outputs: [
                        {
                            pubKey: 'ca80a975d09b84b385e39d779293121b9a7c80942d824d2496f098e3f077ced3',
                            value: 24771038,
                            vout: 1,
                        },
                    ],
                },
                {
                    txid: 'cf655a07ed9b672ecf811ce8b3b69257eecaf0518edd025d7014d7c0177050cb',
                    scanTweak:
                        '03d87550a4ba8ed7dd37df7739a021d6d212e70e8a8e6e9e70249c13a11fd9f412',
                    outputs: [
                        {
                            pubKey: '5af9436eb515871413a913f24290ba003eab7a64b9c62cb72051599a383de8ff',
                            value: 26940,
                            vout: 0,
                        },
                        {
                            pubKey: 'e7a3ab33572f47b074878c749a5f9a5d7e026f10b6ffcab1388e9356c07828fa',
                            value: 4400,
                            vout: 1,
                        },
                    ],
                },
            ],
        },
```

File: packages/core/test/encoding.spec.ts [hunk 3]
```diff
@@ -37,4 +38,22 @@ describe('Encoding', () => {
...
+            transactions: blockData.transactions.map((tx) => ({
+                txid: tx.id,
+                scanTweak: tx.scanTweak,
+                outputs: tx.outputs.map((output) => ({
+                    pubKey: output.pubKey,
+                    value: output.value,
+                    vout: output.vout,
+                })),
+            })),
+        });
```
Comment (theanmolsharma) [inline packages/core/test/encoding.spec.ts:57]:
```suggestion
        expect(parsedBlock).toStrictEqual(blockData.parsedBlock);
```

File: packages/core/test/fixtures/encoding.ts
Comment (theanmolsharma) [inline packages/core/test/fixtures/encoding.ts:1]: We should also add a test for parsing an empty silent block.

**Review by theanmolsharma — CHANGES_REQUESTED** [review]
Nice work!! The `parseSilentBlock` function looks solid 🚀 
Suggested a few nits and test improvements. Make sure you don't create extra commits while addressing the comments.

Before we merge, you **must** sign your commits with a publically verifiable key. [[ref](https://docs.github.com/en/authentication/managing-commit-signature-verification)]

**Review by theanmolsharma — APPROVED** [review]
ACK 6405983eacd667890fa8f382346aa289fc957244
